python/llm/src/ipex_llm/transformers/npu_models/convert_mp.py

```diff
@@ -182,6 +191,11 @@ def optimize_llm(
         from transformers.models.qwen2.modeling_qwen2 import Qwen2ForCausalLM
         from ipex_llm.transformers.npu_models.qwen2_mp import qwen2_casullm_forward
         convert_forward(model, Qwen2ForCausalLM, qwen2_casullm_forward)
+
+        # for Qwen2-7B-Insturct, divide lm_head into 7 parts
+        if model.config.hidden_size == 3584 and model.config.vocab_size == 152064 and \
+                hasattr(model.lm_head, "lm_heads"):
```

Comment:
if model.lm_head is LMHeadLinear

Reply:
Good point, have changed to this condtion.